### PR TITLE
EVG-7120 Add legacy agent distro toggle

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: model/distro/distro.go [hunk 1]
```diff
@@ -347,6 +348,10 @@ func (d *Distro) BinaryName() string {
 
 // ExecutableSubPath returns the directory containing the compiled agents.
 func (d *Distro) ExecutableSubPath() string {
+	arch := d.Arch
+	if d.LegacyAgent {
+		arch = fmt.Sprintf("%s_legacy", arch)
```
Comment (tychoish) [inline model/distro/distro.go]: arch += "_legacy"

File: model/distro/distro.go [hunk 2]
```diff
@@ -38,6 +38,7 @@ type Distro struct {
 	DispatcherSettings    DispatcherSettings      `bson:"dispatcher_settings" json:"dispatcher_settings" mapstructure:"dispatcher_settings"`
 	HostAllocatorSettings HostAllocatorSettings   `bson:"host_allocator_settings" json:"host_allocator_settings" mapstructure:"host_allocator_settings"`
 	DisableShallowClone   bool                    `bson:"disable_shallow_clone" json:"disable_shallow_clone" mapstructure:"disable_shallow_clone"`
+	LegacyAgent           bool                    `bson:"legacy_agent" json:"legacy_agent" mapstructure:"legacy_agent"`
```
Comment (tychoish) [inline model/distro/distro.go]: maybe "UseLegacyAgent"

Comment (bsamek) [reply model/distro/distro.go]: Fixed.

File: model/distro/distro.go (hunk 1 above)
Comment (bsamek) [reply model/distro/distro.go]: Fixed.

**Pull request merged**